test/jdk/java/nio/channels/Channels/ReadXBytes.java

```diff
@@ -138,6 +148,22 @@ public void dataTest(long length, FileCreator c, DataTest f)
         }
     }
 
+    // Runs test against an InputStream where the FileChannel
+    // relationship is obscured.
+    public void dataTestWrappedInputStream(long length, FileCreator c,
+        DataTest f) throws IOException {
+        Path file = c.create(length);
+        try (FileInputStream fis = new FileInputStream(file.toFile());
+             FileInputStream fis2 = new FileInputStream(file.toFile());
+             ReadableByteChannel ch = Channels.newChannel(
+                 new FilterInputStream(fis2) {});
+             InputStream cis = Channels.newInputStream(ch)) {
+            f.test(length, cis, fis);
+        } finally {
+            Files.delete(file);
```

Comment:
How important is exception correctness in the test code? Is it acceptable to leak resources on VM errors?

Reply:
> How important is exception correctness in the test code? Is it acceptable to leak resources on VM errors?

The tests usually run in agentvm mode so if failing tests leak then it may have knock impact to tests that run later in the same VM. So if we can clean up then we should.  Sometimes we have tests that can't reliably cleanup, they can run in othervm mode that is runs just the one test.

Reply:
I meant whether it's acceptable to use a construct like `try (var ch = Channels.newInputStream(new FileInputStream(f)))`. The input stream could leak there, but only in case of VM errors.